graphium/nn/architectures/global_architectures.py

```diff
@@ -1278,6 +1301,95 @@ def set_max_num_nodes_edges_per_graph(self, max_nodes: Optional[int], max_edges:
 
         self.task_heads.set_max_num_nodes_edges_per_graph(max_nodes, max_edges)
 
+    def modify_architecture(self, cfg):
```

Comment:
Pass the parameters here, not the configs.
```suggestion
    def modify_architecture(self, choice, task, add_layers, out_dim, ...):
```

Also, as discussed, `choice` is an ambiguous name